src/gps.c

```diff
@@ -141,7 +141,7 @@ static void *cgps_thread(void *pData) {
         continue;
       }
 
-      if (gps_read(&gpsd_conn) == -1) {
```

Comment:
Hi!

Thanks for your work on Collectd.

Unfortunately, this patch can't be accepted in this form.

The `gpsd` version needs to be checked and conditionals added for both old and new `gps_read()` declarations to be supported.

Reply:
Indeed, you're perfectly right, I sent this patch as a quick fix to buildroot failure but collectd needs to support old versions of gpsd, I will try to work on this and update the PR.